pages/works/[id].tsx

```diff
@@ -0,0 +1,110 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import styles from "@/styles/WorkSinglePage.module.css";
+import { fontSyne, fontSatoshi } from "../_app";
+import works from "@/constants/works.json";
+import Divider from "@/components/base/Divider";
+import Contact from "@/components/general/Contact";
+import InfoItem from "@/components/general/InfoItem";
+
+const WorkSinglePage = () => {
+  const [data, setData] = useState({
+    id: "",
+    title: "",
+    intro: "",
+    info: {
+      client: "",
+      timeline: "",
+      service: "",
+      website: "",
+    },
+    report: {
+      challenge: "",
+      solution: "",
+      result: "",
+    },
+    img: [
+      {
+        src: "",
+        alt: "",
+      },
+      {
+        src: "",
+        alt: "",
+      },
+      {
+        src: "",
+        alt: "",
+      },
+    ],
+  });
+
+  const router = useRouter();
+  const id = router.query.id;
+
+  //which is better? load data here or load it when initiate state
+  useEffect(() => {
+    const workData = works.data.find((item) => item.id === id);
+    if (workData) setData(workData);
+  }, [id]);
+
+  /*I did not put these icons source in data, they are fixed icon based on this page,
+  so it's weird to get them from external data*/
+  type Icons = { [index: string]: string };
+  const icons: Icons = {
+    challenge: "/images/vector.png",
+    solution: "/images/logo.png",
+    result: "/images/video_icon.png",
+  };
+
+  return (
+    <div>
+      <div className={styles["work-single-page__container"]}>
+        <section className={styles["work-single-page__main"]}>
+          <h1 className={`${fontSyne.className} global__heading--large`}>{data.title}</h1>
+          <p className={`${fontSatoshi.className} global__text-xlarge `}>{data.intro}</p>
+        </section>
+
+        <Divider style={{ margin: "0px 48px" }} />
+
+        <section className={styles["work-single-page__aside"]}>
+          {Object.entries(data.info).map(([key, value], index) => (
+            //I think this part is also used in single blog page so I made it a component
+            <InfoItem key={index} title={key} text={value} />
+          ))}
+        </section>
+      </div>
+
+      <div className={styles["work-single-page__container"]}>
+        <section className={styles["work-single-page__aside"]}>
+          {Object.entries(data.report).map(([key, value], index) => (
+            //not sure if I should also make this a component, because it's not so good for reuse
```

Comment:
I dont get it. Why it is not so good for reuse?

Reply:
Like we discussed yesterday, In our project we already have two components for (icon+heading+paragraph): 
ListItem 
![image](https://user-images.githubusercontent.com/54622899/220887907-7eb2dc3c-5ad4-4b86-8b6a-1c6b9ab9db31.png)
and WhatWeDoItem.
![image](https://user-images.githubusercontent.com/54622899/220887813-3c5215c3-212d-4f59-9a50-0f6787a56c54.png)

So, I was curious: should we implemented them by one component? However, you were right that they do have some differences and it might be even more inefficient if we do so. 

As for the new one (icon+heading+paragraph) in this commit, 
![image](https://user-images.githubusercontent.com/54622899/220888292-5951ab9f-4942-4b74-942b-ffb23a841038.png)

It also has some differences from the previous two. I feel I have two options:
1) Make this the third component 
But it will never be used in other places. What's the point to make it a new component? 
If we have three similar components for three similar parts (not the same but not unlike too much) of our website, why don't we just keep them as plain JSX code. 

2) Use a previous component to implement the new one
But it still requires a different styling via props. If this is a good way, why don't we just do this for all (icon+heading+paragraph) but only for this one?

I don't feel either way is logical and self-consistent enough. So I wrote the comment for more discussion.

Reply:
For your question.

1. Yes, it will only be used 3 times for now, and no other places use it. But it will lead to another question. Should we leave all plain JSX for the place but not destruct it into components? Components are not only for dealing with repeat, it also makes your code and file structure easy to read.

2. It can be done in this way either. 

3. You always think too much about these topics. Why do we use agile? Do you really know what agile is? It is not a waterfall project management, it can CHANGE later. If we have bugs, we fix them. If we need to do some enhancement, we do. Agile is a fast pace style of project management, we do it fast.

Remember, if these kinds of questions (such as naming, restructuring, or structure) make you think for more than 5 mins, you do the one way that you simply like more.

Reply:
Thanks! Great explanation.